File: wtplayer/src/WTSurfaceShader.cpp
#include "basetypes.hpp"

#include "WTSurfaceShader.hpp"
#include "WTBitmap.hpp"

WTSurfaceShader::WTSurfaceShader(WT* wt_) :
  WTObject(wt_) {
}

void WTSurfaceShader::setNumLayers(int num_layers){
  APILOG;
}

int WTSurfaceShader::getNumLayers(){
  APILOG;
  return 0;
}

int WTSurfaceShader::getNumPasses(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setDepthOption(int OptionID,
                                     float OptionValue){
  APILOG;
}

void WTSurfaceShader::setLayerSource(int LayerID,
                                     int Source){
  APILOG;
}

int WTSurfaceShader::getLayerSource(int LayerID){
  APILOG;
  return 0;
}

void WTSurfaceShader::setLayerType(int LayerID,
                                   int Type){
  APILOG;
}

int WTSurfaceShader::getLayerType(int LayerID){
  APILOG;
  return 0;
}

void WTSurfaceShader::setTexture(int LayerID,
                                 WTBitmap* Texture_To_Use){
  APILOG;
}

void WTSurfaceShader::setTextureCoordGenMethod(int LayerID,
                                               int TexCoordGenMethod){
  APILOG;
}

int WTSurfaceShader::getTextureCoordGenMethod(int LayerID){
  APILOG;
  return 0;
}

void WTSurfaceShader::setTextureCoordOffsets(int LayerID,
                                             float UOffset,
                                             float VOffset,
                                             float SOffset,
                                             float TOffset){
  APILOG;
}

void WTSurfaceShader::setTextureCoordOffset(int LayerID,
                                            int Coordinate,
                                            float Offset){
  APILOG;
}

float WTSurfaceShader::getTextureCoordOffset(int LayerID,
                                             int Coordinate){
  APILOG;
  return 0;
}

void WTSurfaceShader::setTextureCoordScales(int LayerID,
                                            float UScale,
                                            float VScale,
                                            float SScale,
                                            float TScale){
  APILOG;
}

void WTSurfaceShader::setTextureCoordScale(int LayerID,
                                           int Coordinate,
                                           float Scale){
  APILOG;
}

float WTSurfaceShader::getTextureCoordScale(int LayerID,
                                            int Coordinate){
  APILOG;
  return 0;
}

void WTSurfaceShader::setTextureNumMipMapLevels(int LayerID,
                                                int Depth){
  APILOG;
}

int WTSurfaceShader::getTextureNumMipMapLevels(int LayerID){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometryAmbientColor(unsigned char Red,
                                              unsigned char Green,
                                              unsigned char Blue){
  APILOG;
}

unsigned char WTSurfaceShader::getGeometryAmbientColorRed(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometryAmbientColorGreen(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometryAmbientColorBlue(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometryAmbientOverride(bool Override){
  APILOG;
}

bool WTSurfaceShader::getGeometryAmbientOverride(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometryDiffuseColor(unsigned char Red,
                                              unsigned char Green,
                                              unsigned char Blue,
                                              unsigned char Alpha){
  APILOG;
}

unsigned char WTSurfaceShader::getGeometryDiffuseColorRed(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometryDiffuseColorGreen(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometryDiffuseColorBlue(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometryDiffuseColorAlpha(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometryDiffuseOverride(bool Override){
  APILOG;
}

bool WTSurfaceShader::getGeometryDiffuseOverride(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometryEmissiveColor(unsigned char Red,
                                               unsigned char Green,
                                               unsigned char Blue){
  APILOG;
}

unsigned char WTSurfaceShader::getGeometryEmissiveColorRed(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometryEmissiveColorGreen(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometryEmissiveColorBlue(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometryEmissiveOverride(bool Override){
  APILOG;
}

bool WTSurfaceShader::getGeometryEmissiveOverride(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometrySpecularColor(unsigned char Red,
                                               unsigned char Green,
                                               unsigned char Blue){
  APILOG;
}

unsigned char WTSurfaceShader::getGeometrySpecularColorRed(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometrySpecularColorGreen(){
  APILOG;
  return 0;
}

unsigned char WTSurfaceShader::getGeometrySpecularColorBlue(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometrySpecularPower(float Power){
  APILOG;
}

float WTSurfaceShader::getGeometrySpecularPower(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setGeometrySpecularOverride(bool Override){
  APILOG;
}

bool WTSurfaceShader::getGeometrySpecularOverride(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setProceduralWithString(int LayerID,
                                              char* str){
  APILOG;
}

void WTSurfaceShader::setFrameBufferOperation(int FrameBufferOp){
  APILOG;
}

int WTSurfaceShader::getFrameBufferOperation(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setTextureFiltering(int LayerID,
                                          bool DoFiltering){
  APILOG;
}

bool WTSurfaceShader::getTextureFiltering(int LayerID){
  APILOG;
  return 0;
}

void WTSurfaceShader::setLightingEnabled(bool DoLighting){
  APILOG;
}

bool WTSurfaceShader::getLightingEnabled(){
  APILOG;
  return 0;
}

void WTSurfaceShader::setSortingOption(int Option){
  APILOG;
}

void WTSurfaceShader::setAlphaTestOption(bool Enable,
                                         int OptionValue){
  APILOG;
}

File: wtplayer/javasrc/wildtangent/webdriver/WTSurfaceShader.java
package wildtangent.webdriver;

import wildtangent.webdriver.WTBitmap;
import wildtangent.webdriver.WTObject;

public interface WTSurfaceShader
extends WTObject {
    public void setTextureCoordOffsets(int var1, float var2, float var3, float var4, float var5);

    public void setTextureCoordOffsets(int var1, float var2, float var3, float var4);

    public void setTextureCoordGenMethod(int var1, int var2);

    public int getTextureCoordGenMethod(int var1);

    public void setTextureCoordOffsets(int var1, float var2, float var3);

    public void setGeometryDiffuseOverride(boolean var1);

    public boolean getGeometryDiffuseOverride();

    public void setGeometryEmissiveOverride(boolean var1);

    public boolean getGeometryEmissiveOverride();

    public void setDepthOption(int var1, float var2);

    public void setFrameBufferOperation(int var1);

    public int getFrameBufferOperation();

    public void setAlphaTestOption(boolean var1, int var2);

    public void setAlphaTestOption(boolean var1);

    public void setGeometryAmbientColor(byte var1, byte var2, byte var3);

    public void setGeometryAmbientColor(int var1, int var2, int var3);

    public void setGeometrySpecularColor(byte var1, byte var2, byte var3);

    public void setGeometrySpecularColor(int var1, int var2, int var3);

    public void setGeometryDiffuseColor(byte var1, byte var2, byte var3, byte var4);

    public void setGeometryDiffuseColor(int var1, int var2, int var3, int var4);

    public void setGeometryEmissiveColor(byte var1, byte var2, byte var3);

    public void setGeometryEmissiveColor(int var1, int var2, int var3);

    public void setGeometrySpecularPower(float var1);

    public float getGeometrySpecularPower();

    public void setTextureFiltering(int var1, boolean var2);

    public boolean getTextureFiltering(int var1);

    public byte getGeometryDiffuseColorAlpha();

    public int getNumPasses();

    public void setLightingEnabled(boolean var1);

    public boolean getLightingEnabled();

    public void setTextureCoordScale(int var1, int var2, float var3);

    public float getTextureCoordScale(int var1, int var2);

    public void setTextureNumMipMapLevels(int var1, int var2);

    public int getTextureNumMipMapLevels(int var1);

    public void setLayerSource(int var1, int var2);

    public int getLayerSource(int var1);

    public byte getGeometryAmbientColorGreen();

    public byte getGeometrySpecularColorGreen();

    public void setTexture(int var1, WTBitmap var2);

    public void setSortingOption(int var1);

    public void setTexture(int var1);

    public void setTextureCoordScales(int var1, float var2, float var3, float var4, float var5);

    public void setTextureCoordScales(int var1, float var2, float var3, float var4);

    public void setTextureCoordScales(int var1, float var2, float var3);

    public byte getGeometryDiffuseColorGreen();

    public byte getGeometryEmissiveColorGreen();

    public byte getGeometryAmbientColorRed();

    public byte getGeometrySpecularColorRed();

    public void setLayerType(int var1, int var2);

    public int getLayerType(int var1);

    public byte getGeometryAmbientColorBlue();

    public byte getGeometrySpecularColorBlue();

    public void setTextureCoordOffset(int var1, int var2, float var3);

    public float getTextureCoordOffset(int var1, int var2);

    public byte getGeometryDiffuseColorRed();

    public byte getGeometryEmissiveColorRed();

    public void setNumLayers(int var1);

    public int getNumLayers();

    public void setProceduralWithString(int var1, String var2);

    public byte getGeometryDiffuseColorBlue();

    public byte getGeometryEmissiveColorBlue();

    public void setGeometryAmbientOverride(boolean var1);

    public boolean getGeometryAmbientOverride();

    public void setGeometrySpecularOverride(boolean var1);

    public boolean getGeometrySpecularOverride();
}

